### DISCUSSION TITLE
Can't test using with secure boot because filesystem is read-only

### DISCUSSION THREAD

**Discussion by wsanders** [memtest86plus/memtest86plus#433] [General]
< content deleted >

I think a better way is to just find a way to make grub in my computer’s EFI boot partition boot memtest86+ from disk.

**Comment by debrouxl** [comment]
Using build64/Makefile ( https://github.com/memtest86plus/memtest86plus/blob/main/build64/Makefile ), what if you build esp.img first, sign it manually with your key, and then build memtest.iso ?

I think it's time for us to document the procedure for users to sign memtest86+ binaries with their own signing keys, add the corresponding validation keys in the MOK, and possibly even bake support into the Makefiles for signing with user-provided signing keys.

**Comment by wsanders** [comment]
Thanks for the hints. I think I'll withdraw this request as a WONTFIX. Reasons:

- The memtest86++ RPM works fine with secure boot disabled, the only undocumented step is to copy the EFI file to /boot. 
- There are various descriptions in various places available about signing your own EFI files. (Some of them less incomprehensible than others.)
- For Fedora, should be fairly straightforward (or not!) for the maintainers of memtest86+-7.00 in the repo to sign the EFI file with the Fedora keys, the same way that shimx84.efi is signed. So this can be handed off to the Fedora maintainers since the approach will be different for each distro.
- With the memtest86++ RPM needing to have secure boot disabled, the read-only USB key is just as secure (maybe more secure) than a memtest86+ installed in my /boot partition.

FWIW I Have a dual boot device with Fedora 39 and Windows 11 installed. The device is a Beelink Mini PC that came with the Microsoft KEK and DB9X) keys preinstalled, so both OSes work "out of the box" with secure boot enabled.
